File: internal/storage/storage.go
package storage

import (
	"context"
	"io"
)

type Storage interface {
	Put(ctx context.Context, key string, reader io.Reader, size int64) (string, error)
	URL(key string) string
	Delete(ctx context.Context, key string) error
}

type Config struct {
	Driver string      `json:"driver"`
	Local  LocalConfig `json:"local"`
	S3     S3Config    `json:"s3"`
}

type LocalConfig struct {
	Root string `json:"root"`
	URL  string `json:"url"`
}

type S3Config struct {
	Endpoint  string `json:"endpoint"`
	Bucket    string `json:"bucket"`
	AccessKey string `json:"access_key"`
	SecretKey string `json:"secret_key,omitempty"`
	Region    string `json:"region"`
	Prefix    string `json:"prefix"`
}

File: web/admin/src/api/storage.ts
import client from './client'

export interface StorageConfig { driver: 'local' | 's3'; local: { root: string; url: string }; s3: { endpoint: string; bucket: string; access_key: string; secret_key?: string; region: string; prefix: string } }
export const storageApi = {
  get: () => client.get<StorageConfig>('/backend/v1/storage-config').then((response) => response.data),
  save: (data: StorageConfig) => client.put<StorageConfig>('/backend/v1/storage-config', data).then((response) => response.data),
  test: (data: StorageConfig) => client.post('/backend/v1/storage-config/test', data).then((response) => response.data),
}
